### PR TITLE
Adding convenience method for NNS 

### PR DESCRIPTION
**Pull request by silvanheller**
Inspired by #17, this PR adds a convenience method to the client for NNS search akin to what the SimpleClient has. It also adds a test for this function

### CODE REVIEW

File: cottontaildb_client/cottontaildb_client.py [hunk 1]
```diff
@@ -343,6 +344,41 @@ def ping(self):
...
+        """
+        schema_name = SchemaName(name=schema)
+        entity_name = EntityName(schema=schema_name, name=entity)
+        nnscol = Expression(column=ColumnName(entity=entity_name, name=vector_col))
+
+        distancecol = ColumnName(name='distance')
+        idExpression = Expression(column=ColumnName(name='id'))
+
+        nnslit = Expression(literal=float_vector(*query_vector))
+        fn = FunctionName(name='euclidean')
```
Comment (Spiess) [inline cottontaildb_client/cottontaildb_client.py]: Is it possible the caller provided distance function is never used?
I can't find any reference to it and `euclidean` appears hard-coded.

**Comment by silvanheller** [comment]
Updated & fixed copy-paste errors

File: tests/test_cottontaildb_client.py [hunk 2]
```diff
@@ -110,6 +116,20 @@ def test_query(self):
...
+        query = [0.1, 0.2, 0.4]
+
+        results = self.client.nns(TEST_SCHEMA_STR, TEST_VECTOR_ENTITY_STR, query, vector_col='value')
+        self.assertEqual(len(results), 3, 'unexpected number of rows returned from query')
+        # test with limit
+        results = self.client.nns(TEST_SCHEMA_STR, TEST_VECTOR_ENTITY_STR, query, vector_col='value', limit=1)
+        print(results)
+        self.assertEqual(len(results), 1, 'unexpected number of rows returned from query')
+
+
```
Comment (Spiess) [inline tests/test_cottontaildb_client.py]: PEP 8: E303 too many blank lines

File: tests/test_cottontaildb_client.py [hunk 3]
```diff
@@ -4,14 +4,18 @@
 
 from cottontaildb_client import CottontailDBClient, column_def, Type, Literal, float_vector
 from cottontaildb_client.cottontail_pb2 import Where, AtomicBooleanPredicate, ColumnName, AtomicBooleanOperand, \
-    ComparisonOperator, Expressions, Expression, Projection, IndexType, EntityName, SchemaName
+    ComparisonOperator, Expressions, Expression, Projection, IndexType, EntityName, SchemaName, FunctionName, Function, \
+    Order
```
Comment (Spiess) [inline tests/test_cottontaildb_client.py]: Please remove unused imports

File: cottontaildb_client/cottontaildb_client.py [hunk 4]
```diff
@@ -343,6 +344,41 @@ def ping(self):
...
+
+        nnslit = Expression(literal=float_vector(*query_vector))
+        fn = FunctionName(name=distance)
+        fun = Function(name=fn, arguments=[nnscol, nnslit])
+
+        expression = Expression(function=fun)
+
+        projection_element = Projection.ProjectionElement(alias=distancecol,
+                                                          expression=expression)
+        projection = Projection(op=Projection.ProjectionOperation.SELECT, elements=[projection_element, Projection.ProjectionElement(expression=idExpression)])
```
Comment (Spiess) [inline cottontaildb_client/cottontaildb_client.py]: PEP 8 E501: line too long
Please autoformat to wrap this line

File: cottontaildb_client/cottontaildb_client.py [hunk 5]
```diff
@@ -343,6 +344,41 @@ def ping(self):
...
+        @param limit: maximum number of rows to return
+        @param vector_col: column name where the vector is stored
+        @param id_col: column name where the id is stored
+        """
+        schema_name = SchemaName(name=schema)
+        entity_name = EntityName(schema=schema_name, name=entity)
+        nnscol = Expression(column=ColumnName(entity=entity_name, name=vector_col))
+
+        distancecol = ColumnName(name='distance')
+        idExpression = Expression(column=ColumnName(name=id_col))
```
Comment (Spiess) [inline cottontaildb_client/cottontaildb_client.py]: Please refactor variable names to snake_case. While not as important as `idExpression`, I would prefer to also split names like `distancecol` to `distance_col`.

**Review by Spiess — CHANGES_REQUESTED** [review]
Thanks for the fix and the work it looks great!
I overlooked some minor PEP violations in my first review, after those are fixed we can merge.

**Review by Spiess — APPROVED** [review]

**Pull request merged**